prime-dotnet-webapi/Controllers/HealthAuthoritiesController.cs

```diff
@@ -240,5 +240,62 @@ public async Task<ActionResult> UpdatePharmanetAdministratorContacts(int healthA
             await _healthAuthorityService.UpdateContactsAsync<HealthAuthorityPharmanetAdministrator>(healthAuthorityId, contacts);
             return NoContent();
         }
+
+        // HEAD: api/health-authorities/5/vendor/5
+        /// <summary>
+        /// Checks if the specified vendor is in use by a Health Authority site
+        /// </summary>
+        /// <param name="healthAuthorityId"></param>
+        /// <param name="vendorCode"></param>
+        [HttpHead("{healthAuthorityId}/vendor/{vendorCode}", Name = nameof(IsVendorInUse))]
```

Comment:
I commend you for trying out new Http verbs where they may be better suited than just using plain GET/POST etc. 

However, this is not quite how HEAD works. A HEAD request is semantically the same as a GET without a body; returning only the HEADers that would be returned if the equivalent GET request was performed. So what would you expect a GET to `api/health-authorities/5/vendor/5` to return? I would be very surprised to get back a 404 response code when the resource _exists and is in use by a site_. That does not make sense in the context of a REST API. (Also, the URL should be `api/health-authorities/5/vendor**s**/5`)

Instead, how about `HEAD api/health-authorities/5/vendor/5/sites`? this makes more semantic sense, returning 200OK if there are HA sites using the code, or 404 if there are none using it? 404 is still kinda suspect since you would really expect an empty array rather than 404 if none exist but at least its more in line with expectation.

In fact, I think it would make even more sense if we just did `GET api/health-authorities/5/vendor/5/sites` and it returns a list of site Ids or very basic view models of the sites that are using the vendor, that way we could theoretically message to the user who is using the vendor.

Reply:
I was in fact contemplating the last point of returning the sites that are using the vendor/care type. I only didn't go through with it since the call is barely to prevent a deletion while using that page. I thought it may be better to have a search feature by vendor/care type. But lets have a discussion about that.

Reply:
in the case of this feature, you only would be checking that any Sites/SiteId were returned and blocking the action if any exist.

Reply:
So you mean we still send a list of sites or site ids to front end and block based on whether or not it's empty?

Reply:
Now returning an array of `siteIds`, not currently used outside the logger.